.github/workflows/build.yml

```diff
@@ -29,12 +35,12 @@ jobs:
 
       - name: Upgrade pip
         run: |
-          pip install --constraint=.github/workflows/constraints.txt pip
+          pip install pip
```

Comment:
The removal of constraint makes the behaviour of the workflow susceptible to failure due to braking changes on pip (it happened in the past and that is why I put it). I suggest to keep it. Also, dependabot updates the dependencies on constraints.txt to test them.

Reply:
I figured that was the reason. I'm philosophically opposed to unnecessary pins; a pin should be short-lived only, at least in libraries (shipped applications obviously should be locked). If you insist, fine, but I disagree with the approach.

Reply:
Good point, we can always add when something break, but what I don't like a bit is that you have to track down what version stopped working and when you have many projects to manage pipelines can be broken for some versions... this is unnecessary has a bit of overhead but it is very safe.

Reply:
What I ran into was that the tests were pinned to a surprisingly old version of tox (back on version 3) instead of a modern one. I imagine that's because tox 4 introduced breaking changes to the tox.ini options -- it did! -- but I prefer personally to deal with those breaking changes as soon as they happen rather than waiting til I'm forced to by other things. Pinning to old versions generates tech debt that needs to be paid eventually, and I prefer to be a bit nicer to "Future Chris" :D